### PR TITLE
[receiver/prometheus] set user agent

### PR DESCRIPTION
**Pull request by seankhliao**
**Description:**
Sets the user-agent when the prometheus receiver performs scrapes

**Link to tracking Issue:** #21910

**Testing:** run collector locally and log request:

```
GET /metrics HTTP/1.1
Host: 127.0.0.1:8080
Accept: application/openmetrics-text;version=1.0.0,application/openmetrics-text;version=0.0.1;q=0.75,text/plain;version=0.0.4;q=0.5,*/*;q=0.1
Accept-Encoding: gzip
User-Agent: otelcol-custom/1.0.0
X-Prometheus-Scrape-Timeout-Seconds: 5
```

**Documentation:** changelog

### CODE REVIEW

File: receiver/prometheusreceiver/metrics_receiver.go [hunk 1]
```diff
@@ -281,7 +282,12 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, host component
 	if err != nil {
 		return err
 	}
-	r.scrapeManager = scrape.NewManager(&scrape.Options{PassMetadataInContext: true}, logger, store)
+	r.scrapeManager = scrape.NewManager(&scrape.Options{
```
Comment (MovieStoreGuy) [inline receiver/prometheusreceiver/metrics_receiver.go:285]: Is there any means we can test this within the project?

Comment (seankhliao) [reply receiver/prometheusreceiver/metrics_receiver.go:285]: I was hoping to avoid duplicating most of testComponent to test this

Comment (seankhliao) [reply receiver/prometheusreceiver/metrics_receiver.go:285]: I've added a test

Comment (MovieStoreGuy) [reply receiver/prometheusreceiver/metrics_receiver.go:285]: I wanted to make sure that this doesn't regress again in future.

**Review by MovieStoreGuy — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**